feat(doc): Add the minimum requirements needed for the Exoscale Key in the documentation (#11)
exoscale/cert-manager-webhook-exoscale: README.md

type: Opaque
```

> The IAM role policy of your key should allow at least the following `operation`s for your domain: `list-dns-domains`, `list-dns-domain-records`, `get-dns-domain-record`, `get-operation`, `create-dns-domain-record` and `delete-dns-domain-record`

Reviewer: thank you for the changes!

One last request: can you please remove the `get-operation` from the list(again). We have confirmed that it isn't needed anymore. Sorry for the back and forth.

Author: Done

> The IAM role policy of your key should allow at least the following `operation`s for your domain: `list-dns-domains`, `list-dns-domain-records`, `get-dns-domain-record`, `get-operation`, `create-dns-domain-record` and `delete-dns-domain-record`

Reviewer: ```suggestion
The IAM role policy of your key should allow at least the following `operation`s for your domain: `list-dns-domains`, `list-dns-domain-records`, `get-dns-domain-record`, `create-dns-domain-record` and `delete-dns-domain-record`
```

Here is an example of the minimal policy required for the IAM role:

```json
{
  "default-service-strategy": "deny",
  "services": {
    "dns": {
      "type": "rules",
      "rules": [
        {
          "expression": "resources.dns_domain.unicode_name != \"example.com\"",
          "action": "deny"
        },
        {

>           "expression": "operation in ['list-dns-domains', 'list-dns-domain-records', 'get-dns-domain-record', 'get-operation', 'create-dns-domain-record', 'delete-dns-domain-record']",

Reviewer: ```suggestion
          "expression": "operation in ['list-dns-domains', 'list-dns-domain-records', 'get-dns-domain-record', 'create-dns-domain-record', 'delete-dns-domain-record']",
```

type: Opaque
```

> Your Exoscale Key should have at least the following `operation` for your domain: `list-dns-domain-records`, `get-dns-domain-record`, `list-dns-domains`, `update-dns-domain-record`, `create-dns-domain-record`, `delete-dns-domain-record`

Reviewer: ```suggestion
The IAM role policy of your key should allow at least the following `operation`s for your domain: `list-dns-domain-records`, `get-dns-domain-record`, `list-dns-domains`, `update-dns-domain-record`, `create-dns-domain-record`, `delete-dns-domain-record`
```

Reviewer 2: There is already a note about required permissions [here](https://github.com/exoscale/cert-manager-webhook-exoscale/blob/master/README.md?plain=1#L16).

We can remove the old note now, it referenced the old IAM key type anyway.

> Here is an example of the minimum policy required for the Exoscale Key:

Reviewer: ```suggestion
Here is an example of the minimal policy required for the IAM role:
```

```json
{
  "default-service-strategy": "deny",
  "services": {
    "dns": {
      "type": "rules",
      "rules": [
        {
          "expression": "resources.dns_domain.unicode_name!=\"example.com\"",
          "action": "deny"
        },
        {

>           "expression": "operationin['list-dns-domain-records','get-dns-domain-record','list-dns-domains','update-dns-domain-record','create-dns-domain-record','delete-dns-domain-record']",

Reviewer: ```suggestion
          "expression": "operation in['list-dns-domain-records','get-dns-domain-record','list-dns-domains','update-dns-domain-record','create-dns-domain-record','delete-dns-domain-record']",
```

not your fault, this is a bug where some whitespace is removed when you export an IAM policy from the Portal. We're working on i.

Reviewer: Is there a reason you didn't include the `get-operation` in this list? Does the webhook work without it?
(`get-operation` is listed on line 16)

Reviewer: It looks like you added `update-dns-domain-record` in addition to what we recommend on line 16. Do we really need it?

Reviewer: After checking the source code I'm certain, that we don't need the `update-dns-domain-record` operation. However we do need the `get-operation`.
